File: Backend/src/main/java/org/example/Infrastructure/dto/productDTO/ProductItemDTO.java
package org.example.Infrastructure.dto.productDTO;

import lombok.Data;
import org.example.Infrastructure.dto.commentDTO.CommentItemDTO;

import java.util.List;

@Data
public class ProductItemDTO {
    private int id;
    private String name;
    private String description;
    private String dateCreated;
    private String dateUpdated;
    private double price;
    private List<ProductImageItemDTO> images;
    private int categoryId;
    private String categoryName;
    private List<CommentItemDTO> comments;
    private int numberOfComments;
    private int stars;
}

File: Backend-C#/Infrastructure.MyShop/Models/DTO/ProductDTO/ProductItemDTO.cs
using Infrastructure.MyShop.Models.DTO.CommentDTO;

namespace Infrastructure.MyShop.Models.DTO.ProductDTO
{
    public class ProductItemDTO
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public double Price { get; set; }
        public string DateCreated { get; set; }
        public string DateUpdated { get; set; }
        public int? CategoryId { get; set; }
        public string CategoryName { get; set; }
        public List<ProductImageItemDTO> Images { get; set; }
        public List<CommentItemDTO> Comments { get; set; }
        public int NumberOfComments { get; set; }
        public int Stars { get; set; }
    }
}
